mne/io/base.py

```diff
@@ -92,28 +93,25 @@ def _handle_tmin_tmax(self, tmin, tmax):
             Start time of data to get in seconds.
         tmax : int | float | None
             End time of data to get in seconds.
+        %(include_tmax)s
 
         Returns
         -------
         start : int
             Integer index into data corresponding to tmin.
         stop : int
             Integer index into data corresponding to tmax.
-
         """
         _validate_type(tmin, types=('numeric', None), item_name='tmin',
                        type_name="int, float, None")
         _validate_type(tmax, types=('numeric', None), item_name='tmax',
                        type_name='int, float, None')
+        _validate_type(include_tmax, types=(bool, ), item_name='include_tmax',
+                       type_name='bool')
 
-        # handle tmin/tmax as start and stop indices into data array
-        n_times = self.times.size
-        start = 0 if tmin is None else self.time_as_index(tmin)[0]
-        stop = n_times if tmax is None else self.time_as_index(tmax)[0]
-
-        # truncate start/stop to the open interval [0, n_times]
-        start = min(max(0, start), n_times)
-        stop = min(max(0, stop), n_times)
+        start, stop = np.where(_time_mask(
+            self.times, tmin, tmax, sfreq=self.info['sfreq'],
+            include_tmax=include_tmax))[0][[0, -1]]
```

Comment:
To highlight the code change, to use `_time_mask` internally for time-based selection.